src/network/Connection.ts

```diff
@@ -164,15 +175,22 @@ export class PipelinedWriter extends Writer {
         });
     }
 
-    private handleError(err: any, sentResolvers: OutputQueueItem[]): void {
-        this.error = new IOError(err);
+    private handleError(err: Error, sentResolvers: OutputQueueItem[]): void {
+        const error = this.makeIOError(err);
         for (const item of sentResolvers) {
-            item.resolver.reject(this.error);
+            item.resolver.reject(error);
         }
         for (const item of this.queue) {
-            item.resolver.reject(this.error);
+            item.resolver.reject(error);
```

Comment:
This line is added but not covered, and this is normal. Actually there was no test before my change as well. I did not change the logic of here. Just added this.close() to the end of this function. I don't want to add a test for this because it is not trivial and requires some magic with sinon.